### PR TITLE
feat!: Bump UWP version target to 18362

### LINKED ISSUE #2840: Raise UWP target to 18362
## What would you like to be added:

Update UWP target to version 18362.

## Why is this needed:

Referencing `Microsoft.UI.Xaml` from the SamplesApp to use MUX controls.

After done, enable

- MUX RuntimeTests
- XamlRoot runtime test

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [x] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #2840

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

All UWP project target 17763.

## What is the new behavior?

All UWP project target 18362.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Breaking change

This change removes support for targeting 17763 (October 2018 release of Windows 10) (but it still can be set as "min supported version"). This version is now only supported for Enterprise (as long term) https://en.wikipedia.org/wiki/Windows_10_version_history#Version_1903_(May_2019_Update) . WinUI 2.6 requires targeting 18362 or higher, so developers are essentially forced to target higher anyway.

### CODE REVIEW

File: doc/ReleaseNotes/LegacyReleaseNotes.md [hunk 1]
```diff
@@ -215,7 +215,7 @@
 * [#1559] [#1167] Wasm: make the IsEnabled property inheritable.
 * Full support of pointer events cf. [routed events documentation](../articles/features/routed-events.md)
 * Add support of manipulation events cf. [routed events documentation](../articles/features/routed-events.md)
-* Update CheckBox style to 10.0.17763
+* Update CheckBox style to 10.0.18362
```
Comment (jeromelaban) [inline doc/ReleaseNotes/LegacyReleaseNotes.md:218]: Not needed

File: doc/articles/uno-development/building-uno-ui.md [hunk 2]
```diff
@@ -10,8 +10,8 @@ This article explains how to build Uno.UI locally, for instance if you wish to c
     - `MAUI Preview` (In VS 2022 17.1 Preview 1 or later, for .NET 6 Android/iOS support)
     - `ASP.NET and Web Development`
     - `.NET Core cross-platform development`
-    - `UWP Development`, install all recent UWP SDKs, starting from 10.0.17763 (or above or equal to `TargetPlatformVersion` line [in this file](https://github.com/unoplatform/uno/blob/master/src/Uno.CrossTargetting.props))
-      - Starting from VS 2022, the Windows SDK 10.0.17763 is not available from the installer. Use the [Microsoft Archive](https://developer.microsoft.com/en-us/windows/downloads/sdk-archive/) instead.
+    - `UWP Development`, install all recent UWP SDKs, starting from 10.0.18362 (or above or equal to `TargetPlatformVersion` line [in this file](https://github.com/unoplatform/uno/blob/master/src/Uno.CrossTargetting.props))
+      - Starting from VS 2022, the Windows SDK 10.0.18362 is not available from the installer. Use the [Microsoft Archive](https://developer.microsoft.com/en-us/windows/downloads/sdk-archive/) instead.
```
Comment (jeromelaban) [inline doc/articles/uno-development/building-uno-ui.md:14]: This paragraph can be removed.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_ObjectAnimationUsingKeyFrames.cs [hunk 3]
```diff
@@ -322,7 +322,7 @@ public void When_FirstFrameTimeIsZero_Then_ItsAppliedSyncOnStart()
 		/// </summary>
 		private IDisposable UseDarkTheme() => ThemeHelper.UseDarkTheme();
 
-#if !NETFX_CORE // Disabled on UWP for now because 17763 doesn't support WinUI 2.x
+#if !NETFX_CORE // Disabled on UWP for now because 18362 doesn't support WinUI 2.x
```
Comment (jeromelaban) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_ObjectAnimationUsingKeyFrames.cs:325]: Is this still true ?

File: build/Uno.WinUI.nuspec [hunk 4]
```diff
@@ -614,8 +614,8 @@
 	<file src="uno.winui.win.targets" target="buildTransitive\uap10.0.16299\uno.winui.targets" />
 	<file src="uno.winui.runtime-replace.targets" target="buildTransitive\uap10.0.16299" />
 
-	<file src="uno.winui.win.targets" target="buildTransitive\uap10.0.17763\uno.winui.targets" />
-	<file src="uno.winui.runtime-replace.targets" target="buildTransitive\uap10.0.17763" />
```
Comment (jeromelaban) [inline build/Uno.WinUI.nuspec:618]: We need to keep this block, otherwise apps still referencing 17763 will behave very strangely.

**Review by jeromelaban — COMMENTED** [review]
This needs to be updated:

https://github.com/unoplatform/uno/blob/fb6198da7725a2f999988fa1ea953bcd03a451c5/src/crosstargeting_override.props.sample#L21

File: build/Uno.WinUI.nuspec (hunk 4 above)
Comment (MartinZikmund) [reply build/Uno.WinUI.nuspec:618]: So does it mean there should be both a block for 17763 and for 18362?

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_ObjectAnimationUsingKeyFrames.cs (hunk 3 above)
Comment (MartinZikmund) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_ObjectAnimationUsingKeyFrames.cs:325]: No 👍 !

File: doc/ReleaseNotes/LegacyReleaseNotes.md [hunk 5]
```diff
@@ -590,7 +590,7 @@
 
 ### Features
 * [Wasm] Improve general performance and memory pressure by removing Javascript interop evaluations.
-* Add support for Windows 10 SDK 17763 (1809)
+* Add support for Windows 10 SDK 18362 (1809)
```
Comment (MohammadHadi2031) [inline doc/ReleaseNotes/LegacyReleaseNotes.md:593]: "1809" need to be "1903"

File: build/Uno.WinUI.nuspec (hunk 4 above)
Comment (jeromelaban) [reply build/Uno.WinUI.nuspec:618]: Exactly!

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_ObjectAnimationUsingKeyFrames.cs (hunk 3 above)
Comment (MartinZikmund) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_ObjectAnimationUsingKeyFrames.cs:325]: But the tests do use various Uno-specific APIs, so I have to keep them disabled still.

**Review by jeromelaban — APPROVED** [review]
This needs to be updated:

https://github.com/unoplatform/uno/blob/fb6198da7725a2f999988fa1ea953bcd03a451c5/src/crosstargeting_override.props.sample#L21

**Pull request merged**